tests/fixture/bootstrapmonitor/bootstrap_test_config_test.go

```diff
@@ -0,0 +1,285 @@
+		{
+			name: "container not found",
+			pod: &corev1.Pod{
+				Spec: corev1.PodSpec{},
+			},
+			expectedErr: errContainerNotFound,
+		},
+		{
+			name: "missing network id env var",
+			pod: &corev1.Pod{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name: nodeContainerName,
+						},
+					},
+				},
+			},
+			expectedErr: errInvalidNetworkEnvVar,
+		},
+		{
+			name: "valid configuration without versions and state sync disabled",
+			pod: &corev1.Pod{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  nodeContainerName,
+							Image: imageName,
+							Env: []corev1.EnvVar{
+								{
+									Name:  networkEnvName,
+									Value: networkName,
+								},
+								{
+									Name: chainConfigContentEnvName,
+									// Sets state-sync-enabled:false for the C-Chain
+									Value: "eyJDIjp7IkNvbmZpZyI6ImV5SnpkR0YwWlMxemVXNWpMV1Z1WVdKc1pXUWlPbVpoYkhObGZRPT0iLCJVcGdyYWRlIjpudWxsfX0=",
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedConfig: &BootstrapTestConfig{
+				Network:          networkName,
+				StateSyncEnabled: false,
+				Image:            imageName,
+			},
+		},
+		{
+			name: "valid configuration with valid versions",
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						VersionsAnnotationKey: validVersionsString,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  nodeContainerName,
+							Image: imageName,
+							Env: []corev1.EnvVar{
+								{
+									Name:  networkEnvName,
+									Value: networkName,
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedConfig: &BootstrapTestConfig{
+				Network:          networkName,
+				StateSyncEnabled: true,
+				Image:            imageName,
+				Versions: &version.Versions{
+					Application: "avalanchego/1.11.11",
+					Database:    "v1.4.5",
+					RPCChainVM:  37,
+					Commit:      "5bcfb0fb30cc311adb22173daabb56eae736fac3",
+					Go:          "1.21.12",
+				},
+			},
+		},
+		{
+			name: "invalid configuration due to invalid versions",
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						VersionsAnnotationKey: invalidVersionsString,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name: nodeContainerName,
+							Env: []corev1.EnvVar{
+								{
+									Name:  networkEnvName,
+									Value: networkName,
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedErr: errFailedToUnmarshalAnnoation,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			require := require.New(t)
+
+			config, err := bootstrapTestConfigForPod(test.pod, nodeContainerName)
+			if test.expectedErr != nil {
+				require.ErrorIs(err, test.expectedErr)
+				return
+			}
+			require.NoError(err)
+			require.Equal(test.expectedConfig, config)
```

Comment:
I don't think we technically need this if-statement (applicable to other tests added)? We can explicitly set `expectedErr` and `expectedConfig` to `nil/wantConfigHere` and `errHere/nil` right?